src/openforms/authentication/contrib/digid_eherkenning_oidc/views.py

```diff
@@ -68,7 +72,19 @@ def failure_url(self):
         """
         f = furl(self.success_url)
         f = furl(f.args["next"])
-        f.args[BACKEND_OUTAGE_RESPONSE_PARAMETER] = self.plugin_identifier
+
+        # When the user cancels login we want an explicit error for DigiD authentication
+        # according to the spec.
+        if (
+            oidc_error := self.request.GET.get("error")
+        ) and oidc_error == "access_denied":
+            if (
+                oidc_error_description := self.request.GET.get("error_description")
+            ) and oidc_error_description == "The user cancelled":
+                f.args[DIGID_MESSAGE_PARAMETER] = LOGIN_CANCELLED
```

Comment:
```suggestion
        if self.request.GET.get("error") == "access_denied" and self.request.GET.get("error_description") == "The user cancelled":
            f.args[DIGID_MESSAGE_PARAMETER] = LOGIN_CANCELLED
```

Though it also seems wrong to set the digid message when the user is logging in with eHerkenning, which shares this base class.